src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -194,9 +194,16 @@ public Object eval (DataInstance model, EvaluationContext evalContext) {
         } else if (name.equals("int")) { //non-standard
             assertArgsCount(name, args, 1);
             return toInt(argVals[0]);
-        } else if (name.equals("round")) { // non-standard Excel-style round(value,decimal place)
-            assertArgsCount(name, args, 2);
-            return round(toNumeric(argVals[0]), toNumeric(argVals[1]).intValue());
+        } else if (name.equals("round")) { // Proximate XPath 3.0 and Excel-style round(value,decimal place)
+            final int places;
+            if (args.length == 1) {
+                places = 0;
+             } else {
+                assertArgsCount(name, args, 2);
+                places = toNumeric(argVals[1]).intValue();
+            }
+            // Some locales use ',' instead of '.' as a decimal separator, but not supported by Decimal class.
+            return round(toNumeric(argVals[0].toString().replace(',','.')), places);
```

Comment:
Did you consider doing the replace inside toNumeric?

Reply:
This code has a bug. It bypasses the unpack that toNumeric does.

Reply:
I missed your comment. Thanks for the catch.